File: features/advisor/src/main/java/com/nivasafinance/features/advisorlead/exception/AdvisorLeadMappingOperationException.java
package com.nivasafinance.features.advisorlead.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class AdvisorLeadMappingOperationException extends RuntimeException {

    @Serial
    private static final long serialVersionUID = 1L;

    public AdvisorLeadMappingOperationException(String messageKey, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource));
    }
}

File: features/advisor/src/main/java/com/nivasafinance/features/advisorlead/exception/AdvisorLeadMappingOperationException.kt
package com.nivasafinance.features.advisorlead.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class AdvisorLeadMappingOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
